### PR TITLE
Adds Google authentication and styles comments page with users

### PR DESCRIPTION
**Pull request by cjoshea9**
Adds Google authentication to the website and makes the comment section look better.

### CODE REVIEW

File: portfolio/src/main/webapp/script.js [hunk 1]
```diff
@@ -54,18 +54,52 @@ async function showComments(num=-1) {
...
+        </div>`;
+}
+
+function signOut() {
+  let auth2 = gapi.auth2.getAuthInstance();
+  auth2.signOut().then(function () {
+    location.reload();
+  });
+  const button = document.getElementById('google-buttons');
+  button.innerHTML = `<div class="g-signin2" data-onsuccess="onSignIn" data-theme="dark"></div>`;
```
Comment (Danish21) [inline portfolio/src/main/webapp/script.js:101]: Consider adding element using dom methods like createElement and appendchild:
https://stackoverflow.com/questions/2946656/advantages-of-createelement-over-innerhtml

Here and also above.

File: portfolio/src/main/webapp/photos.html [hunk 2]
```diff
@@ -4,6 +4,8 @@
     <meta charset="UTF-8">
     <title>Photos Page</title>
     <link rel="stylesheet" href="style.css">
+    <script src="https://apis.google.com/js/platform.js" async defer></script>
```
Comment (Danish21) [inline portfolio/src/main/webapp/photos.html:7]: Curious why did you add async and defer? I am not sure what the behavior is when you select both. You probably just want defer.
https://flaviocopes.com/javascript-async-defer/

File: portfolio/src/main/webapp/script.js (hunk 1 above)
Comment (cjoshea9) [reply portfolio/src/main/webapp/script.js:101]: I was wondering which was more efficient. Thank you

Comment (cjoshea9) [reply portfolio/src/main/webapp/script.js:101]: I am actually using it for conditional rendering though depending on whether the user is signed in or signed out. If I use createElement and appendChild, is there a way to conditionally do that?

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -54,18 +54,52 @@ async function showComments(num=-1) {
   for(let i = 0; i<parsedNum; i++) {
     let div = document.createElement('div');
     div.setAttribute('class', 'total-comment');
+    let innerDiv = document.createElement('div');
+    innerDiv.setAttribute('class', 'comment-text');
     let img = document.createElement('img');
-    img.setAttribute('src', '/images/blank.png');
+    img.setAttribute('src', comments[i][1]);
```
Comment (g-harel) [inline portfolio/src/main/webapp/script.js:60]: Using a constant instead of `1` here would make it clearer to the reader (although using an object/map would be even better).

File: portfolio/src/main/webapp/script.js [hunk 4]
```diff
@@ -54,18 +54,52 @@ async function showComments(num=-1) {
   for(let i = 0; i<parsedNum; i++) {
     let div = document.createElement('div');
```
Comment (augerm) [inline portfolio/src/main/webapp/script.js:55]: `let` --> .`const` for these

File: portfolio/src/main/webapp/script.js [hunk 5]
```diff
@@ -54,18 +54,52 @@ async function showComments(num=-1) {
...
-    const response = await fetch('/delete-comments');
-    showComments();
+  const response = await fetch('/delete-comments');
+  showComments();
 }
+
+function onSignIn(googleUser) {
+  const blankProfile = document.getElementById('profile-box');
+  const profile = googleUser.getBasicProfile();
+  let input = document.createElement('input');
```
Comment (augerm) [inline portfolio/src/main/webapp/script.js:82]: `let` --> `const`

File: portfolio/src/main/webapp/script.js [hunk 6]
```diff
@@ -54,18 +54,52 @@ async function showComments(num=-1) {
...
+  const button = document.getElementById('google-buttons');
+  button.innerHTML = `
+        <div id="sign-out" href="#" onclick="signOut()">
+          <img id="google-logo" src="/images/google-logo.jpg">
+          <p>Sign out</p>
+        </div>`;
+}
+
+function signOut() {
+  let auth2 = gapi.auth2.getAuthInstance();
```
Comment (augerm) [inline portfolio/src/main/webapp/script.js:96]: `let` --> `const`

File: portfolio/src/main/webapp/script.js (hunk 1 above)
Comment (Danish21) [reply portfolio/src/main/webapp/script.js:101]: I am not sure I follow. You should be able to take produce the equivalent html using createElement and appendChild, but I feel like I am misunderstanding your question.
